community/blog/2022-06 .NET 6 finally comes to Arcus Templates in v0.8 release/README.md

```diff
@@ -0,0 +1,112 @@
+    }
+}
+```
+
+In Azure Service Bus scenarios, a descriptive `IAzureServiceBusMessageRouter` implementation can be injected into your function which gives you access to the message routing and by extension your registered message handlers.
+
+```csharp
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Abstractions.ServiceBus;
+using Arcus.Messaging.Abstractions.ServiceBus.MessageHandling;
+using Azure.Messaging.ServiceBus;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+
+public class OrderFunction
+{
+    private readonly IAzureServiceBusMessageRouter _messageRouter;
+    private readonly string _jobId;
+
+    public OrderFunction(IAzureServiceBusMessageRouter messageRouter)
+    {
+        _messageRouter = messageRouter;
+        _jobId = Guid.NewGuid().ToString();
+    }
+
+    [FunctionName("order-processing")]
+    public async Task Run(
+        [ServiceBusTrigger("orders", Connection = "ServiceBusConnectionString")] ServiceBusReceivedMessage message,
+        ILogger log,
+        CancellationToken cancellationToken)
+    {
+        log.LogInformation($"C# ServiceBus queue trigger function processed message: {message.MessageId}");
+
+        AzureServiceBusMessageContext context = message.GetMessageContext(_jobId);
+        MessageCorrelationInfo correlationInfo = message.GetCorrelationInfo();
+
+        await _messageRouter.RouteMessageAsync(message, context, correlationInfo, cancellationToken);
+    }
+}
+```
+
+The code described here is an exact implementation of what is available in the new Arcus Azure Functions Service Bus project templates. We have one for both [Azure Service Bus queue](https://templates.arcus-azure.net/features/azurefunctions-servicebus-queue-template) and [Azure Service Bus topic subscription](https://templates.arcus-azure.net/features/azurefunctions-servicebus-topic-template).
+
+## Remove exception details from health report in Web API project template
+The Swagger OpenAPI documentation is by default available in the Web API project template (but can be excluded if needed). Adding Swagger to your API also means that the API models used during request/response will be externalized in the Swagger endpoint JSON. It was by inspecting this Swagger result that we saw a problem with the default application health endpoint that is available in the Web API project template. This endpoint used Microsoft's `HealthReport` model which can include exception details when the application is unhealthy. Generating the Swagger document caused failures because the exception details could contain possible internal type information which resulted in an enormous amount of types that needed to be included in the document. Also from a security perspective, this is not a good approach because it leaks information about the internal system. It could even leak application types that aren't used in the API.
```

Comment:
```suggestion
The OpenAPI documentation is by default available in the Web API project template (but can be excluded if needed). Adding the OpenAPI documentation to your API also means that the API models used during request/response will be externalized in the OpenAPI endpoint JSON. It was by inspecting this OpenAPI documentation that we saw a problem with the default application health endpoint that is available in the Web API project template. This endpoint used Microsoft's `HealthReport` model which can include exception details when the application is unhealthy. Generating the OpenAPI document caused failures because the exception details could contain possible internal type information which resulted in an enormous amount of types that needed to be included in the document. Also from a security perspective, this is not a good approach because it leaks information about the internal system. It could even leak application types that aren't used in the API.
```
I changed Swagger to OpenAPI as Swagger is more the tool than the actual API documentation, see also https://swagger.io/blog/api-strategy/difference-between-swagger-and-openapi/.

Reply:
Aha, ok, thx for this!